### PR TITLE
feat(backend): add inferior Pokemon

### PR DESCRIPTION
**Pull request by 3geek14**
WIP

* existing constants renamed
* Kanto starters added
* intended layout demonstrated

### CODE REVIEW

**Comment by SleepAPI** [comment]
I don't think current commit convention linter accepts optional scope. I found this out the hard way too and it's really unfortunate. I'll create a ticket to investigate an alternative linter, but for now I think you have to drop the scope (backend).

File: backend/src/domain/pokemon/ingredient-pokemon.ts
Comment (3geek14) [reply backend/src/domain/pokemon/ingredient-pokemon.ts:60]: There's some worry about not noticing when it changes, e.g. Onix -> Steelix, but I think the examples are well-known enough that we'll catch them.

Comment (SleepAPI) [reply backend/src/domain/pokemon/ingredient-pokemon.ts:60]: We could add a unit test for the POKEDEX, OPTIMAL_POKEDEX and INFERIOR_POKEDEX constants.
In the test we could get the constants, remove the ingredient and skill percentages and verify the data. 
That way we can still alter the percentages without failing the test, but we ensure that no unexpected changes are made to the constants in terms of ingredient lists, berries etc

File: backend/src/domain/pokemon/time-to-seconds.ts
Comment (3geek14) [reply backend/src/domain/pokemon/time-to-seconds.ts:1]: Moved and added tests. Not amazing tests, but tests nonetheless.

File: backend/src/data/tierlist/cooking/level50/current/curry.json
Comment (3geek14) [reply backend/src/data/tierlist/cooking/level50/current/curry.json:1]: I redid the branch. There's one commit now that changes the tier list data, and it's because it corrects Vigoroth's frequency. The others cleanly don't change the tier list data.

**Comment by SleepAPI** [comment]
Reviewing currently, but to ensure linear history on develop branch, can you pre-squash your commits for me?
So one commit per PR that gets added to develop branch. I realize this is probably also possible to do with Squash and merge, but at that point I have already approved and it's out of my hands.

Allowing squash merging multiple commits also easily ruins conventional commit lint unless carefully handled and would break default branches which is a pain to fix.

To keep consistent with demanding linear history on develop/main I think we have to demand single commit PRs before approving.

File: backend/src/utils/time-utils/time-utils-test.ts
Comment (SleepAPI) [inline backend/src/utils/time-utils/time-utils-test.ts:1]: The naming convention for test files is X.test.ts, rather than X-test.ts as you currently have.
VS code has built-in support for identifying Typescript test files if the aforementioned naming pattern is used.

**Comment by 3geek14** [comment]
Doesn't linear history just require rebasing branches onto `develop` rather than merging them into `develop`? The goal is to avoid making a merge commit on the branch.

I think splitting PRs into multiple commits is a large boon for the ability to review large changes, and I would rather wait until you finish reviewing before I squash. (You can look at the diff one commit at a time, which I think is particularly useful for separating the addition of new mons from the reformatting of existing mons.)

File: backend/src/domain/pokemon/skill-pokemon.ts
Comment (SleepAPI) [inline backend/src/domain/pokemon/skill-pokemon.ts:382]: The Pokemon is called "FLAAFFY" with double A and F

**Review by SleepAPI — CHANGES_REQUESTED** [review]

**Comment by SleepAPI** [comment]
> Doesn't linear history just require rebasing branches onto `develop` rather than merging them into `develop`? The goal is to avoid making a merge commit on the branch.
> 
> I think splitting PRs into multiple commits is a large boon for the ability to review large changes, and I would rather wait until you finish reviewing before I squash. (You can look at the diff one commit at a time, which I think is particularly useful for separating the addition of new mons from the reformatting of existing mons. It also shows that the only change which impacted the tier lists is the modification to Vigoroth's frequency.)

Yeah I agree, it's great for readability on the PR. And indeed the goal is to avoid merge commits on develop/main.
I also have a requirement set on develop/main to require linear history, so hopefully it's impossible to introduce merge commits anyway. 

I guess there's the "fear" of PRs with multiple comments being squash-merged after approval into one messy commit which is pushed to develop. I believe that's possible with the squash merge function, right? After you press squash merge I believe Github asks you how you want to squash your commit messages. If this step is done incorrectly and I have approved already then develop branch would fail build on conventional commit lint and we would have a messy commit in the history.

**Comment by 3geek14** [comment]
I'm not familiar with GitHub to know how those settings work. If there's a way for you to limit who's allowed to click the submit button, then you could squash or not as you see fit and ensure squash commit messages pass linting. If not, I'm fine squashing when you're ready.

**Comment by SleepAPI** [comment]
> I'm not familiar with GitHub to know how those settings work. If there's a way for you to limit who's allowed to click the submit button, then you could squash or not as you see fit and ensure squash commit messages pass linting. If not, I'm fine squashing when you're ready.

Unfortunately Github doesn't let you restrict merge access. I guess Github's policy is that if the code owners approve a PR it's good to go. With this in mind I think we should restrict PRs to one commit per PR, just to keep it consistent for everyone. No risks involved then.

Your latest push looks great and after the squash I'm ready to approve. Amazing job on this @3geek14 👍

**Comment by 3geek14** [comment]
Commits have been squashed.

**Comment by 3geek14** [comment]
Lol, forgot to add Wynaut initially, but that's been fixed now.

**Review by SleepAPI — APPROVED** [review]

**Pull request merged**
